hw/ip/csrng/doc/_index.md

```diff
@@ -0,0 +1,89 @@
+- Support for multiple separate CSRNG instances per IP block.
+    - Each instance has its own internal state, control, reseed counters and IO pins.
+    - The number of RNG instances is set via a module parameter.
+- Software access to a dedicated RNG instance.
+    - One instance, Instance 0, is always accessible from the bus through device registers,
+    - All other instances route to other hardware peripherals (e.g. the key manager, obfuscation engines etc.) and in normal operation these other instances are inaccessible from software.
+    - The IP may be configured to support "debug mode" wherein all instances can be accessed by software.
+      For security reasons this mode may be permanently disabled using one-time programmable (OTP) memory..
+- The IP interfaces with external entropy sources to obtain any required non-determinstic seed material (entropy) and nonces.
+    - Requires an external entropy source which is compliant with NIST SP 800-90B, and which also satisfies the requirements for a PTG.2 class physical non-deterministic random number generator as defined in AIS31.
+    - Dedicated hardware interface with external entropy source satisfies requirements for `get_entropy_input()` interface as defined in SP 800-90A.
+    - When needed, utilizes the `Block_Cipher_df` derivation function (as defined in SP 800-90A) for assembling seed material.
+      This allows the use of entropy sources which are not full-entropy (less than one bit of entropy per bit generated).
+- Also supports the optional use of personalization strings or other application inputs (e.g. OTP memory values) during instantiation.
+- Assuming a continuously-live entropy source, each instance can also optionally be used as a non-determinstic TRNG (true random number generator, also called a non-deterministic random bit generator or NRBG in SP 800-90C).
+    - In this mode, an instance also meets the requirements laid out for a PTG.3 class RNG, the strongest class laid out in AIS31.
+    - Implementation follows the NRBG "Oversampling Construction" approved by SP 800-90C, to meet both CC and FIPS TRNG constructions.
+- In addition to the approved DRNG mode, any instance can also operate in "Fully Deterministic mode", meaning the seed depends entirely on application inputs or personalization strings.
+    - This provides an approved means of key-derivation in FIPS 140-2.
+
+## Description
+
+Though the recommendations in AIS31 are based around broad functional requirements, the recommendations in SP 800-90 are very prescriptive in nature, outlining the exact constructs needed for approval.
+Thus the interface and implementation are largely driven by these explicit constructs, particularly the CTR_DBRG construct.
+
+
+The CSRNG IP consists of four main components:
+1. An AES primitive
+2. The CTR_DRBG state-machine (CTR_DRBG_FSM) which drives the AES primitive, performing the various encryption sequences proscribed in SP 800-90A.
```

Comment:
proscribed -> prescribed
(proscribe means forbid)

Reply:
Right. Good catch, thank you.